modules/nsq/src/it/scala/com/snowplowanalytics/snowplow/enrich/nsq/test/Containers.scala

```diff
@@ -0,0 +1,274 @@
+        lookup2 = NetworkInfo(networkAlias = "nsqlookupd2", broadcastAddress = "nsqlookupd2", httpPort = 4261, tcpPort = 4260),
+        nsqd1 = NetworkInfo(networkAlias = "nsqd", broadcastAddress = "nsqd", httpPort = 4151, tcpPort = 4150),
+        nsqd2 = NetworkInfo(networkAlias = "nsqd2", broadcastAddress = "127.0.0.1", httpPort = 4251, tcpPort = 4250),
+        sourceTopic = "RawEvents",
+        goodDestTopic = "EnrichedEvents",
+        badDestTopic = "BadEnrichedEvents"
+      )
+      _ <- nsqlookupd(network, topology.lookup1)
+      _ <- nsqlookupd(network, topology.lookup2)
+      _ <- nsqd(
+          network,
+          topology.nsqd1,
+          lookupAddress = s"${topology.lookup1.networkAlias}:${topology.lookup1.tcpPort}"
+        )
+      _ <- nsqd(
+          network,
+          topology.nsqd2,
+          lookupAddress = s"${topology.lookup2.networkAlias}:${topology.lookup2.tcpPort}"
+        )
+      _ <- Resource.eval(createTopics[F](blocker, topology))
+      _ <- nsqToNsq(
+          network,
+          sourceAddress = s"${topology.nsqd1.networkAlias}:${topology.nsqd1.tcpPort}",
+          destinationAddress = s"${topology.nsqd2.networkAlias}:${topology.nsqd2.tcpPort}",
+          sourceTopic = topology.goodDestTopic,
+          destinationTopic = topology.goodDestTopic
+        )
+      _ <- nsqToNsq(
+          network,
+          sourceAddress = s"${topology.nsqd1.networkAlias}:${topology.nsqd1.tcpPort}",
+          destinationAddress = s"${topology.nsqd2.networkAlias}:${topology.nsqd2.tcpPort}",
+          sourceTopic = topology.badDestTopic,
+          destinationTopic = topology.badDestTopic
+        )
+      _ <- enrich(network, topology)
+    } yield topology
+
+  private def createTopics[F[_] : Async : ContextShift](blocker: Blocker, topology: NetworkTopology): F[Unit] = {
+    val client = JavaNetClientBuilder[F](blocker).create
+    for {
+      _ <- createTopic(client, topology.sourceTopic, 4151)
+      _ <- createTopic(client, topology.goodDestTopic, 4151)
+      _ <- createTopic(client, topology.badDestTopic, 4151)
+      _ <- createTopic(client, topology.goodDestTopic, 4251)
+      _ <- createTopic(client, topology.badDestTopic, 4251)
+    } yield ()
+  }
+
+  private def createTopic[F[_] : Async : ContextShift](client: Http4sClient[F], topic: String, port: Int): F[Unit] = {
+    val request = Request[F](
+      method = Method.POST,
+      Uri.unsafeFromString(s"http://127.0.0.1:$port/topic/create?topic=$topic")
+    )
+    client.expect(request)
+  }
+
+  private def network[F[_]: Sync](): Resource[F, Network] =
+    Resource.make(
+      Sync[F].delay {
+        Network.newNetwork()
+      }
+    )(
+      n => Sync[F].delay(n.close())
+    )
+
+  private def nsqlookupd[F[_]: Sync](
+    network: Network,
+    networkInfo: NetworkInfo
+  ): Resource[F, JGenericContainer[_]] =
+    Resource.make (
+      Sync[F].delay {
+        val container = FixedHostPortGenericContainer(
```

Comment:
We shouldn't need to use fixed ports, we can use `container.getMappedPort`, like [here](https://github.com/snowplow/enrich/blob/master/modules/kinesis/src/it/scala/com/snowplowanalytics/snowplow/enrich/kinesis/Containers.scala#L70)

Reply:
We have a problem similar to the one explained [here](https://github.com/snowplow/enrich/pull/739#discussion_r1095774056). [nsqd](https://nsq.io/components/nsqd.html) needs to broadcast its port as well. If we don't use fixed ports, it isn't possible to know the assigned port before container started.